cinema/views.py

```diff
@@ -56,3 +76,58 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        movie_id = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+        if movie_id:
+            movie_id = int(movie_id)
+            queryset = queryset.filter(movie_id=movie_id)
+        if date:
+            date = date.split("-")
+            queryset = queryset.filter(
+                show_time__year=int(date[0]),
+                show_time__month=int(date[1]),
+                show_time__day=int(date[2]),
+            )
+        if self.action == "list":
+            queryset = queryset.annotate(
+                tickets_available=F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets")
+            )
+        return queryset
+
+
+class OrderPagination(PageNumberPagination):
+    page_size = 100
+    page_size_query_param = "page_size"
+    max_page_size = 1000
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderPagination
+
+    def get_queryset(self):
+        if self.action in ["list", "retrieve"]:
+            return (
+                Order.objects.filter(user=self.request.user)
+                .select_related("user")
+                .prefetch_related(
+                    "tickets__movie_session__cinema_hall",
+                    "tickets__movie_session__movie",
+                )
+            )
+        return self.queryset
+
+    def get_serializer_class(self):
+        if self.action in ["list", "retrieve"]:
+            return OrderSerializer
+        else:
+            return OrderCreateSerializer
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
```

Comment:
In `perform_create`, ensure that the request user is authenticated before saving the order. This will prevent unauthorized users from creating orders.